api/include/opentelemetry/trace/span_id.h

```diff
@@ -0,0 +1,42 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <cstdint>
+#include <string>
+
+namespace opentelemetry {
+namespace trace {
+
+class SpanId {
+ public:
+  // The size in bytes of the SpanId.
+  static constexpr size_t kSize = 8;
+
+  // An invalid SpanId (all zeros).
+  SpanId() : rep_{0} {}
+
+  // Creates a SpanId by copying the first kSize bytes from the buffer.
+  explicit SpanId(const uint8_t* buf);
```

Comment:
I wish to know what's our thinking on error handling - would this throw when buf is nullptr?

Reply:
IMO passing nullptr should be a SIGSEGV, not a throw. Also we discussed not having exceptions as part of the API.

Reply:
Sorry for not making it clear, I was asking about the input validation. For constructors, do we want to validate input? How about other functions?

Reply:
We can discuss this next week - how do we map the [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/error-handling.md#basic-error-handling-principles) to C++ behaviors.

Reply:
For this particular constructor we shouldn't need to validate input. More generally, this is a topic we should discuss (outside of this PR).